### PR TITLE
Introduce New Chart for the Agent Platform Architecture

### PR DESCRIPTION
**Pull request by frankgreco**
```
---
# Source: superblocks-agent/templates/secret-key.yaml
apiVersion: v1
kind: Secret
metadata:
  name: superblocks
  namespace: default
  labels:
    helm.sh/chart: superblocks-agent-0.23.0
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
    app.kubernetes.io/version: "v0.36.0"
    app.kubernetes.io/managed-by: Helm
type: Opaque
data:
  SUPERBLOCKS_AGENT_KEY: "dGVzdC1hZ2VudC1rZXk=" # test-agent-key
---
# Source: superblocks-agent/templates/secret-tls.yaml
apiVersion: v1
kind: Secret
type: kubernetes.io/tls
metadata:
  name: superblocks-agent-worker-tls
  labels:
    helm.sh/chart: superblocks-agent-0.23.0
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
    app.kubernetes.io/version: "v0.36.0"
    app.kubernetes.io/managed-by: Helm
  # We could use Helm hooks here but there's some weird behvaior.
data:
  tls.crt: <data>
  tls.key: <data>
  tls.ca: <data>
---
# Source: superblocks-agent/templates/service-controller.yaml
apiVersion: v1
kind: Service
metadata:
  name: superblocks
  labels:
    helm.sh/chart: superblocks-agent-0.23.0
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
    app.kubernetes.io/version: "v0.36.0"
    app.kubernetes.io/managed-by: Helm
spec:
  type: ClusterIP
  ports:
  - port: 8020
    targetPort: http
    protocol: TCP
    name: http
  selector:
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
---
# Source: superblocks-agent/templates/deployment-controller.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: superblocks-agent-controller
  labels:
    helm.sh/chart: superblocks-agent-0.23.0
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
    app.kubernetes.io/version: "v0.36.0"
    app.kubernetes.io/managed-by: Helm
    component: controller
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: superblocks-agent
      app.kubernetes.io/instance: superblocks
      component: controller
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 0
      maxSurge: 100%
  template:
    metadata:
      labels:
        app.kubernetes.io/name: superblocks-agent
        app.kubernetes.io/instance: superblocks
        component: controller
    spec:
      terminationGracePeriodSeconds: 86400
      securityContext:
        {}
      containers:
      - name: superblocks-agent-controller
        securityContext:
            {}
        image: "ghcr.io/superblocksteam/agent/controller:v0.36.0"
        imagePullPolicy: IfNotPresent
        ports:
        - name: http
          containerPort: 8020
          protocol: TCP
        - name: worker
          containerPort: 5001
          protocol: TCP
        livenessProbe:
          httpGet:
            path: /health
            port: http
        readinessProbe:
          httpGet:
            path: /health
            port: http
        resources:
            null
        envFrom:
        - secretRef:
          name: superblocks
        env:
        - name: __SUPERBLOCKS_WORKER_ENABLE
          value: "true"
        - name: SUPERBLOCKS_AGENT_PORT
          value: "8020"
        - name: SUPERBLOCKS_AGENT_HOST_URL
          value: "http://localhost:8020/agent"
        - name: SUPERBLOCKS_AGENT_ENVIRONMENT
          value: "*"
        - name: SUPERBLOCKS_WORKER_TLS_CA_FILE
          value: /etc/config/tls/tls.ca
        - name: SUPERBLOCKS_WORKER_TLS_CERT_FILE
          value: /etc/config/tls/tls.crt
        - name: SUPERBLOCKS_WORKER_TLS_KEY_FILE
          value: /etc/config/tls/tls.key
        - name: SUPERBLOCKS_WORKER_PORT
          value: "5001"
        - name: SUPERBLOCKS_AGENT_INTERNAL_HOST
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        volumeMounts:
        - name: tls
          mountPath: /etc/config/tls
      volumes:
      - name: tls
        secret:
          secretName: superblocks-agent-worker-tls
---
# Source: superblocks-agent/templates/deployment-worker.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: superblocks-agent-worker-language
  labels:
    helm.sh/chart: superblocks-agent-0.23.0
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
    app.kubernetes.io/version: "v0.36.0"
    app.kubernetes.io/managed-by: Helm
    component: worker
    fleet: language
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: superblocks-agent
      app.kubernetes.io/instance: superblocks
      component: worker
      fleet: language
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 0
      maxSurge: 100%
  template:
    metadata:
      labels:
        app.kubernetes.io/name: superblocks-agent
        app.kubernetes.io/instance: superblocks
        component: worker
        fleet: language
    spec:
      terminationGracePeriodSeconds: 86400
      securityContext:
        null
      containers:
      - name: superblocks-agent-worker
        securityContext:
          null
        image: "ghcr.io/superblocksteam/agent/worker:v0.36.0"
        imagePullPolicy: IfNotPresent
        resources:
          {}
        envFrom:
        - secretRef:
          name: superblocks
        env:
        - name: SUPERBLOCKS_WORKER_TLS_CA_FILE
          value: /etc/config/tls/tls.ca
        - name: SUPERBLOCKS_WORKER_TLS_CERT_FILE
          value: /etc/config/tls/tls.crt
        - name: SUPERBLOCKS_WORKER_TLS_KEY_FILE
          value: /etc/config/tls/tls.key
        - name: SUPERBLOCKS_WORKER_PLUGINS
          value: "javascript,python"
        - name: SUPERBLOCKS_WORKER_CONCURRENCY
          value: "1"
        volumeMounts:
        - name: tls
          mountPath: /etc/config/tls
      volumes:
      - name: tls
        secret:
          secretName: superblocks-agent-worker-tls
---
# Source: superblocks-agent/templates/deployment-worker.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: superblocks-agent-worker-not-language
  labels:
    helm.sh/chart: superblocks-agent-0.23.0
    app.kubernetes.io/name: superblocks-agent
    app.kubernetes.io/instance: superblocks
    app.kubernetes.io/version: "v0.36.0"
    app.kubernetes.io/managed-by: Helm
    component: worker
    fleet: not-language
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: superblocks-agent
      app.kubernetes.io/instance: superblocks
      component: worker
      fleet: not-language
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 0
      maxSurge: 100%
  template:
    metadata:
      labels:
        app.kubernetes.io/name: superblocks-agent
        app.kubernetes.io/instance: superblocks
        component: worker
        fleet: not-language
    spec:
      terminationGracePeriodSeconds: 86400
      securityContext:
        null
      containers:
      - name: superblocks-agent-worker
        securityContext:
          null
        image: "ghcr.io/superblocksteam/agent/worker:v0.36.0"
        imagePullPolicy: IfNotPresent
        resources:
          {}
        envFrom:
        - secretRef:
          name: superblocks
        env:
        - name: SUPERBLOCKS_WORKER_TLS_CA_FILE
          value: /etc/config/tls/tls.ca
        - name: SUPERBLOCKS_WORKER_TLS_CERT_FILE
          value: /etc/config/tls/tls.crt
        - name: SUPERBLOCKS_WORKER_TLS_KEY_FILE
          value: /etc/config/tls/tls.key
        - name: SUPERBLOCKS_WORKER_PLUGINS
          value: "!javascript,python"
        volumeMounts:
        - name: tls
          mountPath: /etc/config/tls
      volumes:
      - name: tls
        secret:
          secretName: superblocks-agent-worker-tls
```

### CODE REVIEW

File: helm/templates/pdb-controller.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.podDisruptionBudget -}}
+{{- if .Values.controller.podDisruptionBudget -}}
```
Comment (bruce-y) [inline helm/templates/pdb-controller.yaml:1]: Should we also include a worker pdb?

File: helm/templates/hpa-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.controller.autoscaling.enabled }}
```
[the remark this answers is not in the document]
Comment (frankgreco) [reply helm/templates/hpa-controller.yaml:1]: I guess it doesn't hurt to have the based cpu/memory hpas and then optimize the metrics later.

**Review by bruce-y — APPROVED** [review]

File: docker/compose.yaml [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
+  worker:
+    container_name: worker
+    image: ${SUPERBLOCKS_DOCKER_WORKER_REPOSITORY:-ghcr.io/superblocksteam/agent/worker}:${SUPERBLOCKS_DOCKER_WORKER_TAG:-latest}
+    pull_policy: ${SUPERBLOCKS_DOCKER_IMAGE_PULL_POLICY:-always}
+    environment: *environment
+    ports:
+    - '${SUPERBLOCKS_WORKER_METRICS_PORT:-9090}:${SUPERBLOCKS_WORKER_METRICS_PORT:-9090}'
+  controller:
+    container_name: controller
+    image: ${SUPERBLOCKS_DOCKER_CONTROLLER_REPOSITORY:-ghcr.io/superblocksteam/agent/controller}:${SUPERBLOCKS_DOCKER_CONTROLLER_TAG:-latest}
```
Comment (taha-au) [inline docker/compose.yaml:27]:
```suggestion
    image: ${SUPERBLOCKS_DOCKER_CONTROLLER_REPOSITORY:-ghcr.io/superblocksteam/agent-controller}:${SUPERBLOCKS_DOCKER_CONTROLLER_TAG:-latest}
```

**Review by taha-au — CHANGES_REQUESTED** [review]
LGTM other than the requested changes

**Review by taha-au — APPROVED** [review]
Pushed commit to address requested changes

**Pull request merged**